sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala

```diff
@@ -193,11 +208,18 @@ class RocksDBFileManager(
       version: Long,
       useColumnFamilies: Boolean = false): StateStoreChangelogReader = {
     val changelogFile = dfsChangelogFile(version)
-    if (useColumnFamilies) {
-      new StateStoreChangelogReaderV2(fm, changelogFile, codec)
-    } else {
-      new StateStoreChangelogReaderV1(fm, changelogFile, codec)
+
+    val changelogVersion = getChangelogVersion(useColumnFamilies)
```

Comment:
We'll need to rely on the version info rather than the outer information, but I understand this is problematic as v1 does not have the version info and we have to put some workaround logic to determine whether there is a version info (v1) or not. OK to revisit this when we are going to add v3.

Reply:
Yea - since we don't have for v1 - just rely on external args here and then verifying that we are reading the right decided/expected version. Added a note in the code as well.